detection_rules/main.py

```diff
@@ -299,52 +299,48 @@ def update_lock_versions(rule_ids):
 @root.command('kibana-diff')
 @click.option('--rule-id', '-r', multiple=True, help='Optionally specify rule ID')
 @click.option('--branch', '-b', default='master', help='Specify the kibana branch to diff against')
-def kibana_diff(rule_id, branch):
+@click.option('--threads', '-t', type=click.IntRange(1), default=50, help='Number of threads to use to download rules')
+def kibana_diff(rule_id, branch, threads):
```

Comment:
default is 50 threads? maybe use multiprocessing and the number of cores instead

Reply:
I think 50 is fine for threads. These are all network connections and so shouldn't add a ton of overhead, so threading makes more sense than multiproc IMO.

Reply:
Use threadpool 👍